### PR TITLE
updates to production example

### PR DESCRIPTION
**Pull request by lea-hagen**
This makes several changes for the production run example.  The most significant is making an uber-wrapper for the production run (`beast_production_wrapper.py`), which does basically everything in the [standard workflow](http://beast.readthedocs.io/en/latest/workflow.html#). It also uses a datamodel template to automatically fill in the filters and file names (the user does the rest before running), to account for the case where there are many fields with different names and filter combinations.  I re-organized a few pieces of code (`run_beast_production`, `reorder_beast_results_spatial`, `condense_beast_results_spatial`) to be callable as functions, though basically all of their code is the same.  In addition, `setup_batch_beast_fit` now returns a dictionary with information about which fields need running, so that `beast_production_wrapper` knows whether to move along with the workflow or wait until the fitting is complete.

### CODE REVIEW

**Comment by karllark** [comment]
Please update the documentation as appropriate as part of this pull request.

**Comment by lea-hagen** [comment]
Documentation is updated!  Let me know if there's anything unclear.

**Comment by karllark** [comment]
There are errors in the documentation build.  Check them out at:
https://travis-ci.org/BEAST-Fitting/beast/jobs/419853035

Remember, you can generate the documentation on your own computer using the 'python setup.py build_docs' command.  Useful for catching errors before pushing them to github.  :-)

**Comment by lea-hagen** [comment]
That's weird, I did successfully build the docs on my computer before pushing the changes!  I'll poke at it a little more to see if I can get it to work here too.

File: docs/workflow.rst [hunk 1]
```diff
@@ -22,22 +22,83 @@ template is the 'metal_production' subdirectory in beast/examples.
...
   .. code:: shell
 
      $ ln -s /location/beast/beast/ beast
 
-Datamodel.py
-============
+
+
+Datamodel_template.py
+===============
```
Comment (karllark) [inline docs/workflow.rst]: This one of the lines that is giving a warning.  The '===' have to have the same number of characters as the label above it.

File: docs/workflow.rst [hunk 2]
```diff
@@ -22,22 +22,83 @@ template is the 'metal_production' subdirectory in beast/examples.
...
 Before running the BEAST, you will need to modify this file to specify
 the required parameters for generating models and fitting data.
-These parameters are described in the beast :ref:`setup documentation <beast_setup_datamodel>`.
+These parameters are described in the beast :ref:`setup documentation
+<beast_setup_datamodel>`.  The fields for project, obsfile, astfile,
+filters, and basefilters will be filled in by beast_production_wrapper.py.
 
+****
+Beast_production_wrapper.py
+****
```
Comment (karllark) [inline docs/workflow.rst]: Both the above and below '**' will also give a warning.

**Comment by karllark** [comment]
Ready for merging now?

**Comment by lea-hagen** [comment]
yes!

**Pull request merged**